addons/l10n_account_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,348 @@
+                # As we handle this on the wizard itself, we can't rely on the onchange to update this.
+                line.custom_user_currency_id = line.currency_id
+
+    # -----------------------
+    # CRUD, inherited methods
+    # -----------------------
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """ When selecting an outstanding account manually, we want it to be reconcilable.
+        Similarly to what is done on a journal when setting the outstanding of a payment method, we'll thus mark the
+        account as reconcilable if it makes sense.
+        """
+        #
+        # EXTEND account
+        for vals in vals_list:
+            if vals.get('l10n_account_wth_outstanding_account_id'):
+                account = self.env['account.account'].browse(vals['l10n_account_wth_outstanding_account_id'])
+                if not account.reconcile and account.account_type not in ('asset_cash', 'liability_credit_card', 'off_balance'):
+                    account.reconcile = True
+        return super().create(vals_list)
+
+    # ----------------
+    # Business methods
+    # ----------------
+
+    def _create_payment_vals_from_wizard(self, batch_result):
+        # EXTEND account
+        # Applies the withholding logic to the payment record when creating it.
+        payment_vals = super()._create_payment_vals_from_wizard(batch_result)
+
+        if self.l10n_account_wth_net_amount < 0:
+            raise UserError(_('The net amount cannot be negative.'))
+
+        if not self.l10n_account_wth_line_ids or not self.l10n_account_wth_withhold_tax:
+            return payment_vals  # Nothing to do if we are not working with withholding taxes.
+
+        # Get the account set on the company; raise an error if not set.
+        tax_base_account = self.company_id.l10n_account_wth_tax_base_account_id.id
+        if not tax_base_account:
+            action = self.env.ref('account.action_account_config')
+            raise RedirectWarning(
+                _('To register withholding taxes, the "Withholding Tax Base Account" must be set in the settings.'),
+                action.id,
+                _('Accounting Settings')
+            )
+        # The payment amount is the base amount set in the wizard, minus the sum of the withholding line amounts.
+        payment_vals['amount'] = self.l10n_account_wth_net_amount
+        # Ensure that we get a journal entry
+        if self.l10n_account_wth_outstanding_account_id:
+            payment_vals['outstanding_account_id'] = self.l10n_account_wth_outstanding_account_id.id
+        # We need to process the withholding here as the records are transient, and it won't be done by the payment.
+        # /!\ including currency conversion if needed.
+        conversion_rate = self.env['res.currency']._get_conversion_rate(
+            self.currency_id,
+            self.company_id.currency_id,
+            self.company_id,
+            self.payment_date,
+        )
```

Comment:
Use `_convert` instead, it is clearer to see from/to which currency we are converting + the rounding is part of the conversion anyway

Reply:
I did it this way to avoid calling _convert in a loop.
But sure, I can change it :ok_hand:

Reply:
`_convert` is cached